input/fsh/SD_AdverseEvent.fsh

```diff
@@ -29,25 +29,50 @@ Description: "Profile of adverse event, using Common Terminology Criteria (CTC).
-* extension[CTCAEGrade].valueCodeableConcept MS
-* extension[CTCAEGrade].valueCodeableConcept.coding ^short = "Adverse Event Grade"
-* extension[CTCAEGrade].valueCodeableConcept.coding ^definition = "The grade of the adverse event, determined by CTCAE criteria. The code '0' representing no adverse event may be used to provide positive confirmation that the clinician assessed or considered this particular AE, although the absence of an adverse event is generally not reportable."
+* extension contains 
+    CTCAEGrade named grade 1..1 MS and
+    AdverseEventExpectation named expectation 0..1 MS and
+    AdverseEventResolvedDate named resolvedDate 0..1 MS
+* extension[CTCAEGrade] ^short = "Adverse Event Grade"
+* extension[CTCAEGrade] ^definition = "The grade of the adverse event, determined by CTCAE criteria. The code '0' representing no adverse event may be used to provide positive confirmation that the clinician assessed or considered this particular AE, although the absence of an adverse event is generally not reportable."
+* extension[AdverseEventExpectation] ^short = "Adverse Event Expectation"
+* extension[AdverseEventExpectation] ^definition = "A determination if the adverse event is or is not one whose nature and severity have been previously observed, identified in nature, severity, or frequency, and documented in the investigator brochure, investigational plan, protocol, current consent form, scientific publication, or in other relevant and reliable document."
+* extension[AdverseEventResolvedDate] ^short = "Adverse Resolved Date"
+* extension[AdverseEventResolvedDate] ^definition = "The date (and time) when the adverse event ends or returns to baseline (NCI Thesaurus)."
 // ------Seriousness------
 * seriousness 0..1 MS
 * seriousness from AdverseEventSeriousnessVS (required)
 // ------Causality------
 * suspectEntity and suspectEntity.instance and suspectEntity.causality and suspectEntity.causality.assessment and suspectEntity.causality.assessment.text MS
 * suspectEntity.causality.assessment from AdverseEventRelatednessVS (required)
 
-
-Extension: CTCAEGrade
-Id: mcode-ctcae-grade
-Title: "CTC Adverse Event Grade"
-Description: "The grade associated with the severity of an adverse event, using CTCAE criteria. See https://ctep.cancer.gov/protocolDevelopment/electronic_applications/ctc.htm"
+RuleSet: AdverseEventExtensionPreamble
 * ^status = #draft
 * ^experimental = true
 * ^context[0].type = #element
 * ^context[0].expression = "AdverseEvent"
+
+
+Extension: CTCAEGrade
+Id: ctcae-grade
+Title: "CTC Adverse Event Grade"
+Description: "The grade associated with the severity of an adverse event, using CTCAE criteria. The code '0' representing no adverse event may be used to provide positive confirmation that the clinician assessed or considered this particular AE, although the absence of an adverse event is generally not reportable. See https://ctep.cancer.gov/protocolDevelopment/electronic_applications/ctc.htm"
```

Comment:
Do you want to mention `https://ctep.cancer.gov/protocolDevelopment/electronic_applications/ctc.htm` in `* extension[CTCAEGrade] ^definition`?

(It would be nice if FSH provided a way to avoid duplicating the text between the element definition and the Extension description.)